src/Lean/Meta/Tactic/ElimInfo.lean

```diff
@@ -178,15 +178,79 @@ def addCustomEliminator (declName : Name) (attrKind : AttributeKind) (induction
   let e ← mkCustomEliminator declName induction
   customEliminatorExt.add e attrKind
 
-builtin_initialize
+/--
+Registers a custom eliminator for the `induction` tactic.
+
+Whenever the types of the targets in an `induction` call matches a custom eliminator, it is used
+instead of the recursor. This can be useful for redefining the default eliminator to a more useful
+one.
+
+Example:
+```lean example
+structure Three where
+  val : Fin 3
+
+example (x : Three) (p : Three → Prop) : p x := by
+  induction x
+  -- val : Fin 3 ⊢ p ⟨val⟩
+
+@[induction_eliminator, elab_as_elim]
+def Three.myRec {motive : Three → Sort u}
+    (zero : motive ⟨0⟩) (one : motive ⟨1⟩) (two : motive ⟨2⟩) :
+    ∀ x, motive x
+  | ⟨0⟩ => zero | ⟨1⟩ => one | ⟨2⟩ => two
+
+example (x : Three) (p : Three → Prop) : p x := by
+  induction x
+  -- ⊢ p ⟨0⟩
+  -- ⊢ p ⟨1⟩
+  -- ⊢ p ⟨2⟩
+```
+
+`@[cases_eliminator]` works similarly for the `cases` tactic.
```

Comment:
I think that this is a very good example, but I'd rather see it in the reference manual with a link from here. That helps keep the code up to date as Lean changes, and it's easier to read in a browser than in a hover. As with the others, I can do this if it's too much to ask.